Automattic/wp-calypso
# Global Styles on Personal A/B: Try dynamic import of `useExperiment`

## Proposed Changes

Addresses feedback from @jessie-ross and imports the `loadExperimentAssignment` hook into the `useSiteGlobalStylesStatus` hook with a dynamic import so it doesn't produce any error in SSR contexts.

See paYJgx-3yE-p2#comment-3703

## Testing Instructions

- Open the Calypso live link below in a incognito window.
- Observe the network requests with your browser devtools.
- Go to `/themes` while logged out.
- Make sure it renders correctly.
- Make sure there are no requests to the `/experiments/0.1.0/assignments/calypso` endpoint (because you're logged out).
- While logged out, go to `/setup/newsletter` and click on the "Launch my newsletter" button.
- Log in.
- After logging in, once you're in the "It begins with a name" screen, make sure there is a request to the `/experiments/0.1.0/assignments/calypso` endpoint (because you're logged in and you didn't select/create a site yet).
- Go to `/themes` while logged in and pick a site.
- Make sure there are no requests to the `/experiments/0.1.0/assignments/calypso` endpoint (because you're logged in and you selected a site, so the experiment assignment is fetched from the `sites/:site/global-styles/status` endpoint instead).
